region/maridia/outer/Crab Hole.json

```diff
@@ -2114,21 +2706,149 @@
     {
       "id": 96,
       "link": [4, 1],
-      "name": "Come In Shinecharging, Leave Shinecharged",
+      "name": "Come In Shinecharging, Leave Shinecharged (Gravity, Less Runway)",
       "entranceCondition": {
         "comeInShinecharging": {
-          "length": 2,
-          "openEnd": 0
+          "length": 1,
+          "openEnd": 1
         }
       },
       "requires": [
         "Gravity",
-        "canShinechargeMovement",
+        "canShinechargeMovementComplex",
         {"shineChargeFrames": 100}
       ],
       "exitCondition": {
         "leaveShinecharged": {}
       },
+      "unlocksDoors": [
+        {"types": ["super"], "requires": []},
+        {"types": ["missiles", "powerbomb"], "requires": ["never"]}
+      ],
```

Comment:
Where these doors are capless, I think we have been treating them as not-lockable (by ammo).  But someone could always put a door cap on it.  So they should also add the unlock requirements themselves in that case?

Reply:
Yeah, the way the tests are currently written, it allows unlocksDoors on those but doesn't require them. I think it could make sense to change that and start including them consistently. I think the only case that wouldn't make sense to include would be the sand transitions. 

There's a bunch of cases across many rooms, so maybe will defer that to a separate PR later.